server.js

```diff
@@ -42,9 +42,56 @@ server.use('/dev', proxy(DEVSERVER))
 server.listen(PORT, () => console.log(`listening on port ${PORT}`))
 
 
+const read_active_filter = ({on_filter}) => {
+    // Step 1: See if there is an active context
+    const tw_context = readline.createInterface({
+        input: spawn_tw(['_get', 'rc.context']).stdout
+    })
+    let context = ''
+    tw_context.on('line', (line) => {
+        if (!context && line) context = line
+    })
+    tw_context.on('close', (line) => {
+        // Step 2: If there is an active context, get its filter.
+        //         If not, get the filter for the "next" report.
+        let cmd = ['_get']
+        if (context)
+            cmd.push('rc.context.' + context)
+        else
+            cmd.push('rc.report.next.filter')
+
+        const tw_filter = readline.createInterface({
+            input: spawn_tw(cmd).stdout
+        })
+
+        let filter = ''
+        tw_filter.on('line', (line) => {
+            if (!filter && line) filter = line
+        })
+        tw_filter.on('close', (line) => {
+            // Execute callback with obtained filter
+            on_filter('(' + filter + ')')
+        })
+    })
+}
+
+
+const export_filtered_tasks = (filter, callbacks) => {
+    // If a filter is already passed, don't bother checking taskwarrior's
```

Comment:
With the use case I had for the filter in the query, this is not correct -- both the TW filters and the given filter should be applied. However, since the UI is not using the `filter` query parameter, maybe it would be simpler to drop it for now. Currently it's an unused feature polluting the code, so I'd argue it's better to just get rid of it.